packages/authentication/src/metadata.ts

```diff
@@ -0,0 +1,15 @@
+// Copyright IBM Corp. 2013,2017. All Rights Reserved.
+// Node module: loopback
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import * as assert from 'assert';
+import 'reflect-metadata';
+
+export function authenticate(controllerClass: Object, methodName: string) {
+  Reflect.defineMetadata('loopback:authenticate', true, controllerClass, methodName);
```

Comment:
I will handle these 2 test cases which you want me to add. But @bajtos just a question here, your comment mentions `@authenticate()` signature whereas I have `@authenticate` in my implementation. So If you expect me to change the signature I will have to change my implementation too.